contracts/exchange-wrappers/TestExchangeWrapper.sol

```diff
@@ -0,0 +1,150 @@
+/*
+
+    Copyright 2018 dYdX Trading Inc.
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+
+*/
+
+pragma solidity 0.5.1;
+
+import { ExchangeWrapper } from "../interfaces/ExchangeWrapper.sol";
+import { TokenInteract } from "../lib/TokenInteract.sol";
+
+
+/**
+ * @title TestExchangeWrapper
+ * @author dYdX
+ *
+ * An ExchangeWrapper for testing
+ */
+contract TestExchangeWrapper is
+    ExchangeWrapper
+{
+    using TokenInteract for address;
+
+    // ============ Constants ============
+
+    // arbitrary address to send the tokens to (they are burned)
+    address constant EXCHANGE_ADDRESS = address(0x1);
+
+    // ============ Structs ============
+
+    struct Order {
+        address originator;
+        address makerToken;
+        address takerToken;
+        uint256 makerAmount;
+        uint256 takerAmount;
+    }
+
+    // ============ ExchangeWrapper functions ============
+
+    function exchange(
+        address tradeOriginator,
+        address receiver,
+        address makerToken,
+        address takerToken,
+        uint256 requestedFillAmount,
+        bytes calldata orderData
+    )
+        external
+        returns (uint256)
+    {
+        Order memory order = parseData(orderData);
+
+        require(
+            order.originator == tradeOriginator,
```

Comment:
Why are you passing all this info in in the order if it's already given in `exchange`? Just so you can validate everything else actually sends through the bytes correctly?